scripts/missions/asa/04_Sugar-Coated_Directive.lua

```diff
@@ -0,0 +1,277 @@
+                        local sealCount = mission:getVar(player, 'SealCount')
+                        mission:setVar(player, 'SealCount', sealCount + 1)
+                    end,
+                }
+            },
+        },
+
+        [xi.zone.CLOISTER_OF_TREMORS] =
+        {
+            ['Earth_Protocrystal'] =
+            {
+                onTrigger = function(player, npc)
+                    local asaPrimal = mission:getVar(player, 'Titan')
+
+                    if player:hasKeyItem(xi.ki.DOMINAS_AMBER_SEAL) and asaPrimal == 0 then
+                        mission:setVar(player, 'Titan', 1)
+                        return mission:messageSpecial(tremorsID.text.POWER_STYMIES, xi.keyItem.DOMINAS_AMBER_SEAL)
+                    elseif player:hasKeyItem(xi.ki.DOMINAS_AMBER_SEAL) and asaPrimal == 2 then
+                        mission:setVar(player, 'Titan', 3)
+                        return mission:progressEvent(2)
+                    end
+                end,
+
+                onEventFinish =
+                {
+                    [2] = function(player, csid, option, npc)
+                        player:delKeyItem(xi.ki.DOMINAS_AMBER_SEAL)
+                        player:addKeyItem(xi.ki.AMBER_COUNTERSEAL)
+                        player:messageSpecial(tremorsID.text.ATTACH_SEAL, xi.ki.DOMINAS_AMBER_SEAL)
+                        player:messageSpecial(tremorsID.text.KEYITEM_OBTAINED, xi.ki.AMBER_COUNTERSEAL)
+                        local sealCount = mission:getVar(player, 'SealCount')
+                        mission:setVar(player, 'SealCount', sealCount + 1)
+                    end,
+                }
+            },
+        },
+
+        [xi.zone.QUFIM_ISLAND] =
+        {
+            ['Trodden_Snow'] =
+            {
+                onTrigger = function(player, npc)
+                    --ASA 4 CS: Triggers With At Least 3 Counterseals.
+                    local completedSeals =
+                        (player:hasKeyItem(xi.ki.AMBER_COUNTERSEAL)    and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.AZURE_COUNTERSEAL)    and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.CERULEAN_COUNTERSEAL) and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.EMERALD_COUNTERSEAL)  and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.SCARLET_COUNTERSEAL)  and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.VIOLET_COUNTERSEAL)   and 1 or 0)
+
+                    if completedSeals >= 3 then
+                        mission:setVar(player, 'SealCount', completedSeals)
+                        return mission:progressEvent(45)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [45] = function(player, csid, option, npc)
+                    local completedSeals = mission:getVar(player, 'SealCount')
+
+                    -- Calculate Reward
+                    if completedSeals == 3 then
```

Comment:
keyed table for num seals completed would work here as well to reduce duplicated code